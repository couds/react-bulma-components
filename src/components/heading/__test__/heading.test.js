import React from 'react';
import renderer from 'react-test-renderer';
import Heading from '..';

describe('Heading component', () => {
  it('Should exist', () => {
    expect(Heading).toMatchSnapshot();
  });
  it('Should have title classname', () => {
    const component = renderer.create(
      <Heading>
        Test <a>Give me</a>
      </Heading>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should concat classname in props with Bulma classname', () => {
    const component = renderer.create(
      <Heading className="other-class this-is-a-test">
        <p>Default</p>
      </Heading>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should use inline styles', () => {
    const component = renderer.create(
      <Heading style={{ height: 250 }}>
        <p>Default</p>
      </Heading>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should be a subtitle with size rendered as P', () => {
    const component = renderer.create(
      <Heading size={3} subtitle renderAs="p">
        Subtitle
      </Heading>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should not be spaced because is subtitle', () => {
    const component = renderer.create(
      <Heading spaced subtitle renderAs="p">
        Subtitle
      </Heading>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should be spaced', () => {
    const component = renderer.create(
      <Heading spaced renderAs="p">
        Subtitle
      </Heading>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
