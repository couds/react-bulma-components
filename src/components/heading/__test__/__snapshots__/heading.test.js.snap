// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading component Should be a subtitle with size rendered as P 1`] = `
<p
  className="subtitle is-3"
>
  Subtitle
</p>
`;

exports[`Heading component Should be spaced 1`] = `
<p
  className="title is-spaced"
>
  Subtitle
</p>
`;

exports[`Heading component Should concat classname in props with Bulma classname 1`] = `
<h1
  className="other-class this-is-a-test title"
>
  <p>
    Default
  </p>
</h1>
`;

exports[`Heading component Should exist 1`] = `[Function]`;

exports[`Heading component Should have title classname 1`] = `
<h1
  className="title"
>
  Test 
  <a>
    Give me
  </a>
</h1>
`;

exports[`Heading component Should not be spaced because is subtitle 1`] = `
<p
  className="subtitle"
>
  Subtitle
</p>
`;

exports[`Heading component Should use inline styles 1`] = `
<h1
  className="title"
  style={
    Object {
      "height": 250,
    }
  }
>
  <p>
    Default
  </p>
</h1>
`;
